tracing-subscriber/src/registry/mod.rs

```diff
@@ -241,17 +351,70 @@ where
         })
     }
 
+    /// Returns an iterator over all parents of this span, starting with this span,
+    /// ordered from leaf to root.
+    ///
+    /// The iterator will first return the span, then the span's immediate parent,
+    /// followed by that span's parent, and so on, until it reaches a root span.
+    ///
+    /// ```rust
+    /// use tracing_subscriber::registry::{Registry, LookupSpan};
+    /// tracing::subscriber::with_default(tracing_subscriber::registry(), || {
+    ///     let _root = tracing::info_span!("root").entered();
+    ///     let _child = tracing::info_span!("child").entered();
+    ///     let leaf = tracing::info_span!("leaf");
+    ///     tracing::dispatcher::get_default(|dispatch| {
+    ///         let registry = dispatch.downcast_ref::<Registry>().unwrap();
+    ///         let leaf_scope = registry.span(&leaf.id().unwrap()).unwrap().scope();
+    ///         assert_eq!(
+    ///             leaf_scope.map(|span| span.name()).collect::<Vec<&str>>(),
+    ///             vec!["leaf", "child", "root"]
+    ///         );
+    ///     });
+    /// });
+    /// ```
+    ///
+    /// If the opposite order (from the root to this span) is desired, calling [`Scope::from_root`] on
+    /// the returned iterator reverses the order.
+    ///
+    /// ```rust
+    /// use tracing_subscriber::registry::{Registry, LookupSpan};
+    /// tracing::subscriber::with_default(tracing_subscriber::registry(), || {
+    ///     let _root = tracing::info_span!("root").entered();
+    ///     let _child = tracing::info_span!("child").entered();
+    ///     let leaf = tracing::info_span!("leaf");
+    ///     tracing::dispatcher::get_default(|dispatch| {
+    ///         let registry = dispatch.downcast_ref::<Registry>().unwrap();
+    ///         let leaf_scope = registry.span(&leaf.id().unwrap()).unwrap().scope();
+    ///         assert_eq!(
+    ///             leaf_scope.from_root().map(|span| span.name()).collect::<Vec<&str>>(),
+    ///             vec!["root", "child", "leaf"]
+    ///         );
+    ///     });
+    /// });
+    /// ```
+    pub fn scope(&self) -> Scope<'a, R> {
+        Scope {
+            registry: self.registry,
+            next: Some(self.id()),
+        }
+    }
+
     /// Returns an iterator over all parents of this span, starting with the
     /// immediate parent.
     ///
     /// The iterator will first return the span's immediate parent, followed by
     /// that span's parent, followed by _that_ span's parent, and so on, until a
     /// it reaches a root span.
+    #[deprecated(
+        note = "equivalent to `self.parent().into_iter().flat_map(SpanRef::scope)`, but consider whether excluding self is actually intended"
```

Comment:
also, let's add the version
```suggestion
        note = "equivalent to `self.parent().into_iter().flat_map(SpanRef::scope)`, but consider whether excluding self is actually intended",
        since = "0.2.19",
```